addons/stock_picking_batch/models/stock_picking.py

```diff
@@ -27,6 +39,28 @@ def _compute_picking_count(self):
             for record in self:
                 record[field] = count.get(record.id, 0)
 
+    @api.model
+    def _get_batch_group_by_keys(self):
+        return ['batch_group_by_partner', 'batch_group_by_destination', 'batch_group_by_src_subloc', 'batch_group_by_dest_subloc']
+
+    @api.constrains(lambda self: self._get_batch_group_by_keys() + ['auto_batch'])
+    def _validate_auto_batch_group_by(self):
+        group_by_keys = self._get_batch_group_by_keys()
+        for picking_type in self:
+            if not picking_type.auto_batch:
+                continue
+            if not any(self[key] for key in group_by_keys):
+                raise ValidationError(_("If the Automatic Batches feature is enabled, at least one 'Group by' option must be selected."))
+
+    @api.model
+    def update_incoming_outgoing_types(self):
+        picking_type_ids = self.env['stock.picking.type'].search([('code', 'in', ['incoming', 'outgoing'])])
+        for picking_type in picking_type_ids:
+            picking_type.write({
+                'auto_batch': True,
+                'batch_group_by_partner': True
+            })
```

Comment:
Not sure you should put that in the python file. You could do it directly in the data file

Reply:
But it needs to update all existing incoming/outgoing picking types (as multiple warehouses may have been created by the time batches are enabled and thus this code triggered), doesn't it ?

Reply:
Done